nncf/experimental/common/pruning/operations.py

```diff
@@ -668,15 +689,59 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
 
         node.data['output_mask'] = output_mask
 
-
 class StopMaskForwardPruningOp(BasePruningOp):
-    @classmethod
-    def accept_pruned_input(cls, node: NNCFNode) -> bool:
-        return False
-
     @classmethod
     def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
                          tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
         input_masks = get_input_masks(node, graph)
         cls.invalidate_masks(input_masks)
         node.data['output_mask'] = None
+
+class ExpandAsPruningOp(BasePruningOp):
+    @classmethod
+    def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
+                         tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> None:
+        input_edges = graph.get_input_edges(node)
+        assert len(input_edges) == 2, 'expand should always have 2 inputs'
+        input_to_expand = input_edges[0].from_node
+        mask = input_to_expand.data.get('output_mask')
+        propagated_mask = None
+        if mask is not None:
+            nncf_logger.warning(f"expand_as is applied to the node with propagation mask. Currently, it's not supported "
+                                "and mask is invalidated. node_name={node.node_name}")
+            mask.invalidate_groups()
+        input_to_get_shape = input_edges[1].from_node
+        mask: PropagationMask = input_to_get_shape.data.get('output_mask')
+        if mask is not None:
+            target_shape = input_edges[1].tensor_shape
+            source_shape = input_edges[0].tensor_shape
+            for dim, groups in mask.dim_groups_map.items():
+                if target_shape[dim] == source_shape[dim]:
+                    nncf_logger.warning(f"expand_as takes the shape from the node with propagation mask and pruning "
+                                        "dimension in the mask matches the dimension in the expanded input. Currently, "
+                                        "it's not supported and mask is invalidated. node_name={node.node_name}")
+                    for group in groups:
+                        group.invalidate()
+            # TODO: assume that expand_as is on constant path that does not affect pruning, otherwise pruning of
```

Comment:
please, format TODO regarding the style guide.

Reply:
Discussed in person, will be resolved in https://github.com/openvinotoolkit/nncf/pull/1704